report_migration/report_data.sql

```diff
@@ -0,0 +1,131 @@
+    equip_torso_id INT,
+    equip_legs_id INT,
+    equip_boots_id INT,
+    equip_cape_id INT,
+    equip_hands_id INT,
+    equip_weapon_id INT,
+    equip_shield_id INT,
+    equip_ge_value BIGINT,
+    report_sighting_id INT DEFAULT NULL,
+    report_location_id INT DEFAULT NULL,
+    report_gear_id INT DEFAULT NULL
+);
+
+-- Step 2: Insert data into temp_batch (manually adjust startId, endId, and batch_size as needed)
+
+INSERT INTO temp_batch (ID, created_at, reportedID, reportingID, region_id, x_coord, y_coord, z_coord,
+                        timestamp, manual_detect, on_members_world, on_pvp_world, world_number,
+                        equip_head_id, equip_amulet_id, equip_torso_id, equip_legs_id, equip_boots_id,
+                        equip_cape_id, equip_hands_id, equip_weapon_id, equip_shield_id, equip_ge_value)
+SELECT ID, created_at, reportedID, reportingID, region_id, x_coord, y_coord, z_coord,
+       timestamp, manual_detect, on_members_world, on_pvp_world, world_number,
+       equip_head_id, equip_amulet_id, equip_torso_id, equip_legs_id, equip_boots_id,
+       equip_cape_id, equip_hands_id, equip_weapon_id, equip_shield_id, equip_ge_value
+FROM Reports
+WHERE ID > :startId AND ID <= :endId -- Python Variables
+ORDER BY ID ASC;
+
+-- Step 3: Update temp_batch with existing foreign keys from report_sighting
+UPDATE temp_batch AS tb
+LEFT JOIN report_sighting AS rs
+ON tb.reportingID = rs.reporting_id AND tb.reportedID = rs.reported_id AND tb.manual_detect = rs.manual_detect
+SET tb.report_sighting_id = rs.report_sighting_id;
+
+
+
+-- Debug: Check temp_batch after update
+-- SELECT * FROM temp_batch WHERE report_sighting_id IS NULL;
+
+-- Step 4: Insert missing rows into report_sighting
+INSERT INTO report_sighting (reporting_id, reported_id, manual_detect)
+SELECT DISTINCT reportingID, reportedID, manual_detect
+FROM temp_batch
+WHERE report_sighting_id IS NULL
+ORDER BY reportingID, reportedID, manual_detect;
+
+-- Debug: Check rows inserted into report_sighting
+-- SELECT * FROM report_sighting ORDER BY report_sighting_id DESC LIMIT 10;
+
+-- Step 5: Update temp_batch with newly inserted report_sighting IDs
+UPDATE temp_batch AS tb
+JOIN report_sighting AS rs
+ON tb.reportingID = rs.reporting_id AND tb.reportedID = rs.reported_id AND tb.manual_detect = rs.manual_detect
+SET tb.report_sighting_id = rs.report_sighting_id
+WHERE tb.report_sighting_id IS NULL;
+
+-- Debug: Verify report_sighting_id is updated
+-- SELECT * FROM temp_batch;
+
+-- Step 6: Insert missing rows into report_gear
+INSERT INTO report_gear (equip_head_id, equip_amulet_id, equip_torso_id, equip_legs_id,
+                         equip_boots_id, equip_cape_id, equip_hands_id, equip_weapon_id, equip_shield_id)
+SELECT DISTINCT equip_head_id, equip_amulet_id, equip_torso_id, equip_legs_id,
+                equip_boots_id, equip_cape_id, equip_hands_id, equip_weapon_id, equip_shield_id
+FROM temp_batch
+WHERE report_gear_id IS NULL
```

Comment:
if you want to commit to the update table idea you'll have to check every table first